### PR TITLE
Work on deflaking Storage integration test on mobile, particularly the large file download/pause/resume.

### PR DESCRIPTION
**Pull request by jonsimantov**
This test adds a convenience method RunFlakyBlock to firebase_test_framework that allows you to run a block multiple times, with exponential backoff until it succeeds (or fails a few times). This allows you to perform tests that have a chance of failure, such as uploading/downloading large files.

### CODE REVIEW

**Review by DellaBitta — APPROVED** [review]
Approved, with comments.

File: storage/integration_test/src/integration_test.cc [hunk 1]
```diff
@@ -624,7 +628,16 @@ class StorageListener : public firebase::storage::Listener {
...
-    controller->Resume();
+    LogDebug("Resuming");
+    resume_succeeded_ = controller->Resume();
+    if (!resume_succeeded_) {
+      // Resume failed, try it one more time.
+      ProcessEvents(1000);
+      resume_succeeded_ = controller->Resume();
+    }
+    if (resume_succeeded_) {
+      LogDebug("Resume succeeded");
```
[the remark this answers is not in the document]
Comment (jonsimantov) [reply storage/integration_test/src/integration_test.cc:638]: Better yet...

File: storage/integration_test/src/integration_test.cc [hunk 2]
```diff
@@ -672,51 +687,93 @@ TEST_F(FirebaseStorageTest, TestLargeFilePauseResumeAndDownloadCancel) {
...
-
-    while (controller.bytes_transferred() == 0) {
+  struct Context {
+    firebase::storage::StorageReference* ref;
+    const std::string* test_file;
+    size_t test_file_size;
+  } context;
+  context.ref = &ref;
+  context.test_file = &kLargeTestFile;
+  context.test_file_size = kLargeFileSize;
```
[the remark this answers is not in the document]
Comment (jonsimantov) [reply storage/integration_test/src/integration_test.cc:699]: Added a note about how safe it is.

File: storage/integration_test/src/integration_test.cc [hunk 3]
```diff
@@ -672,51 +687,93 @@ TEST_F(FirebaseStorageTest, TestLargeFilePauseResumeAndDownloadCancel) {
...
-    // Ensure the Controller is valid now that we have associated it with an
-    // operation.
-    ASSERT_TRUE(controller.is_valid());
-
-    while (controller.bytes_transferred() == 0) {
+  struct Context {
+    firebase::storage::StorageReference* ref;
+    const std::string* test_file;
+    size_t test_file_size;
+  } context;
```
[the remark this answers is not in the document]
Comment (jonsimantov) [reply storage/integration_test/src/integration_test.cc]: Ugh, so wordy.

File: storage/integration_test/src/integration_test.cc (hunk 1 above)
[the remark this answers is not in the document]
Comment (DellaBitta) [reply storage/integration_test/src/integration_test.cc:638]: Better yet?

**Review by DellaBitta — APPROVED** [review]

File: storage/integration_test/src/integration_test.cc (hunk 1 above)
[the remark this answers is not in the document]
Comment (jonsimantov) [reply storage/integration_test/src/integration_test.cc:638]: I changed it to use RetryFlakyBlock. :)

**Pull request merged**